### PR TITLE
[pvr] Allow usage of addon channel setting functions

### PR DESCRIPTION
**Pull request by AlwinEsch**
In channel manager (is if supported on a addon) in list at end a entry with "Add channel", further are on channels where possible two new context menu items with "Edit" and "Delete" on all other not visible.

With this change becomes following functions on addon interface used again:

DialogAddChannel
DialogChannelSettings
DeleteChannel
MoveChannel
RenameChannel

It comes with it the new support flag for addons with "bSupportsChannelSettings" which must be set true if above functions are supported. With this it is related to pull request #6326 to reduce amount of API version changes.

Is required for my radio addon where everything becomes handled inside KODI.

Due to pull request #6093 I'm not sure if this way is OK, but to use again own buttons make now no seens for me, is only used by one addon.

### CODE REVIEW

**Comment by xhaggi** [comment]
@AlwinEsch did you mixed this changes with the undelete recording stuff?

**Comment by AlwinEsch** [comment]
Yes is temporary cherry picked to bring them together if possible to reduce amount of Version changes

**Comment by xhaggi** [comment]
If so I would prefer that you separate the version bump to a new PR and merge it after this and the other one is in. @opdenkamp any objections?

**Comment by AlwinEsch** [comment]
Also if possible with #6328, there is also a gui lib version change.

**Comment by AlwinEsch** [comment]
Is no problem for me to change the version to a separate one. But then together in a short time.

**Comment by xhaggi** [comment]
thanks way better to review now ;)

**Comment by AlwinEsch** [comment]
Is updated, has leaved the addon function names, there are more and need a separate pull request.
Also is if supported on one addon a "New Channel" button present.

**Comment by opdenkamp** [comment]
not a big fan of the add-on opening dialogs, but it'll have to do. for now ;-)

re API changes from multiple PRs: to ensure that we don't end up with a broken build on some platform, we have to do it like this:
- update the add-ons to include all API changes that are to be merged in https://github.com/opdenkamp/xbmc-pvr-addons/tree/master
- create a new PR here that combines all PRs that are changing the API
- include a version bump of the add-ons in this new PR
- have jenkins build it and merge it

**Comment by AlwinEsch** [comment]
Thanks @opdenkamp I update it then and make everything together in #6367. 

I can also create a pull request on all to cmake separeted addons.

**Comment by AlwinEsch** [comment]
One question @opdenkamp, @xhaggi mean to rename also the addon function names here https://github.com/AlwinEsch/kodi/commit/0a76e24d5d7538e21eec61072ae3bc850e076e8b#commitcomment-9643049 , how you see this?

File: xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp [hunk 1]
```diff
@@ -616,6 +692,14 @@ void CGUIDialogPVRChannelManager::Update()
     SET_CONTROL_LABELS(SPIN_EPGSOURCE_SELECTION, 0, &labels);
   }
 
+  m_clientsWithSettingsList = g_PVRClients->GetClientsSupportingChannelSettings(m_bIsRadio);
+  if (!m_clientsWithSettingsList.empty())
+    m_bAllowNewChannel = true;
+
+  CGUIButtonControl *btnShowNewChannel = (CGUIButtonControl*) GetControl(BUTTON_NEW_CHANNEL);
+  if (btnShowNewChannel)
+    btnShowNewChannel->SetVisible(m_bAllowNewChannel);
```
Comment (xhaggi) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp]: As I said on IRC you should use the macro's SET_CONTROL_HIDDEN() and  SET_CONTROL_VISIBLE().

```
if (m_bAllowNewChannel)
  SET_CONTROL_VISIBLE(BUTTON_NEW_CHANNEL);
else
  SET_CONTROL_HIDDEN(BUTTON_NEW_CHANNEL);
```

Comment (AlwinEsch) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp]: Upss, has in the night a small difficult in my mind, searched for the wrong name of it and thought is not used anymore ;-). Becomes fixed :-D

Comment (xhaggi) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp]: Thanks ;)

**Comment by AlwinEsch** [comment]
Is transfered to #6367 and this becomes closed.

@xhaggi, with SET_CONTROL is done there.

**Pull request closed without merging**